os/kernel/log_dump/log_dump.c

```diff
@@ -109,10 +146,75 @@ int log_dump_init(void)
 	return LOG_DUMP_OK;
 }
 
-void log_dump_read_init(void)
+unsigned char *last_comp_block;
+size_t last_comp_block_size;
+int last_comp_block_ptr;
+bool last_block_compression;
+
+int log_dump_read_wake(void)
+{
+	last_block_compression = true;
+	struct log_dump_msg_s msg;
+	int ret;
+
+	msg.comp = true;
+	msg.ch = 'c';
+	ret = mq_send(log_dump_get_mqfd(), (const char *)&msg, sizeof(struct log_dump_msg_s), 100);
+	if (ret < 0) {
+		lldbg("failed mq_send ret = %d\n", ret);
+		return ERROR;
+	}
+
+	while (last_block_compression) ;
+	return 0;
+}
+
+int log_dump_read_init(void)
 {
 	read_node = (struct log_dump_chunk_s *)sq_peek(&log_dump_chunks);	/* reset the read node to head */
-	current_ptr = 0;							/* reset the read pointer */
+
+	if(read_node == NULL) {
+		lldbg("empty read node\n");
+		return LOG_DUMP_MEM_FAIL;
+	}
+
+	if (read_node->comp_head < 0) {
+		current_ptr = curbytes;
+	} else {
+		current_ptr = read_node->comp_head;
+	}
+
+	last_comp_block_size = CONFIG_LOG_DUMP_CHUNK_SIZE;
+
+#if defined(CONFIG_LIBLZMA)
+	last_comp_block = (unsigned char *)kmm_malloc(4 + CONFIG_LOG_DUMP_CHUNK_SIZE + LZMA_PROPS_SIZE);
+#elif defined(CONFIG_LIB_MINIZ)
+	last_comp_block = (unsigned char *)kmm_malloc(4 + CONFIG_LOG_DUMP_CHUNK_SIZE);
+#endif
+	if (last_comp_block == NULL) {
+		lldbg("memory allocation failure\n");
+		return LOG_DUMP_MEM_FAIL;
+	}
+
+#if defined(CONFIG_LIB_LZMA)
+	int ret = LzmaCompress(&last_comp_block[4 + LZMA_PROPS_SIZE], &last_comp_block_size, uncomp_array, uncomp_curbytes, &last_comp_block[4], &propsSize, 0, 1 << 13, -1, -1, -1, -1, 1);
+#elif defined(CONFIG_LIB_MINIZ)
+	int ret = mz_compress(&last_comp_block[4], &last_comp_block_size, uncomp_array, uncomp_curbytes);
+#endif
```

Comment:
I think we should support compress function in compression module, and compress module uses miniz or lzma internally. I think it's proper layering. How do you think?

Reply:
any modules can use compression libraries directly.  compression module is more tuned for binary compression

Reply:
If it is, it should be in binary manager. But it isn't. We can use it in many modules and it's not good at usability to add both in the app like below.
```
#ifdef config_lib_lzma
   use lzma function
#else
   use minin function
#endif
```

Reply:
modified to include compress logic in compression module